vm-setup/roles/common/tasks/main.yml

```diff
@@ -6,21 +6,13 @@
 - name: Define vm_nodes if not already defined
   when: generate_vm_nodes
   block:
-    - name: Generate vm_nodes for "{{num_masters}}" masters
+    - name: Generate vm_nodes for "{{num_nodes}}" nodes
       set_fact:
         vm_nodes: "{{vm_nodes|default([]) + [
-                     {'name': ironic_prefix + 'master_%s'|format(item),
-                      'flavor': 'master',
+                     {'name': ironic_prefix + 'node_%s'|format(item),
+                      'flavor': 'node',
                       'virtualbmc_port': virtualbmc_base_port+item}]}}"
-      loop: "{{ range(0, num_masters|int)|list }}"
-    
-    - name: Generate vm_nodes for "{{num_workers}}" workers
-      set_fact:
-        vm_nodes: "{{vm_nodes|default([]) + [
-                     {'name': ironic_prefix + 'worker_%s'|format(item),
-                      'flavor': 'worker',
-                      'virtualbmc_port': virtualbmc_base_port+num_masters|int+item} ]}}"
-      loop: "{{ range(0, num_workers|int)|list }}"
+      loop: "{{ range(0, num_nodes|int)|list }}"
```

Comment:
Could we add a loop here to iterate over the defined flavors, and set facts for the counts by looking for a num_$flavorname variable, with defaults defined e.g 0 when the variable isn't set?